front/lib/resources/vault_resource.ts

```diff
@@ -0,0 +1,202 @@
+import type {
+  ACLType,
+  LightWorkspaceType,
+  ModelId,
+  Result,
+} from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type {
+  Attributes,
+  CreationAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { VaultModel } from "@app/lib/resources/storage/models/vaults";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import { getResourceIdFromSId, makeSId } from "@app/lib/resources/string_ids";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface VaultResource extends ReadonlyAttributesType<VaultModel> {}
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class VaultResource extends BaseResource<VaultModel> {
+  static model: ModelStatic<VaultModel> = VaultModel;
+
+  constructor(model: ModelStatic<VaultModel>, blob: Attributes<VaultModel>) {
+    super(VaultModel, blob);
+  }
+
+  static async makeNew(blob: CreationAttributes<VaultModel>) {
+    const vault = await VaultModel.create(blob);
+
+    return new this(VaultModel, vault.get());
+  }
+
+  get sId(): string {
+    return VaultResource.modelIdToSId({
+      id: this.id,
+      workspaceId: this.workspaceId,
+    });
+  }
+
+  static modelIdToSId({
+    id,
+    workspaceId,
+  }: {
+    id: ModelId;
+    workspaceId: ModelId;
+  }): string {
+    return makeSId("vault", {
+      id,
+      workspaceId,
+    });
+  }
+
+  static async fetchWorkspaceVaults(
```

Comment:
`nit` it should be using the verb `list`.
`fetch` => One row
`list` => Many rows